### DISCUSSION TITLE
View system installation progress

### DISCUSSION THREAD

**Discussion by 1392273211** [cobbler/cobbler#3615] [Q&A]
I configured anamon_enabled in /etc/cobbler/settings: 1 and then restarted the httpd and cobblerd services. Sometimes I can see the progress of the client's installation of the system, but sometimes not. In addition to this method, there are other ways to view the system in real time. Is the installation progress?

**Comment by SchoolGuy** [comment]
Good Morning! Normally your system has a graphical or serial output this is nothing that can be utilized by Cobbler but you can manually watch the output. Automatically reporting the systems status during installation can be done by using `cobbler status`.

Since I just realized that this command is undocumented I have opened the following issue for this: #3616

**Reply by 1392273211** [reply]
I have executed the cobbler status command, but after the client is installed successfully, I cannot see whether the installation is successful. I have to check it manually. If I install hundreds of nodes, it is impossible to check them one by one. Is there any solution?

**Reply by SchoolGuy** [reply]
Your templates need to use the `autoinstall_start` and `autoinstall_done` snippets so that Cobbler is notified that the system has started and finished installation. Otherwise `cobbler status` will not pick an installation up correctly.

You can parse the output of `cobbler status` and check for systems installing longer than anticipated. This is normally a sign that something has gone wrong. It is impossible for Cobbler to check the actual status of the installation due to the sheer amount of different installers, things that can go wrong, and the complexity of physically installing a system that would be able to monitor something like this.

**Reply by 1392273211** [reply]
When you say that the template needs to use the autoinstall_start and autoinstall_done fragments, do you mean adding them in the ks file or configuring them in cobbler? Is there any documentation reference?

**Reply by SchoolGuy** [reply]
The documentation about the templates is not written because it is expected that admins understand them by themselves so far.

You can look at how the `default.ks` template handles this.

**Reply by 1392273211** [reply]
The content in default.ks of cobbler 2.8 is empty
```
# this file intentionally left blank
# admins: edit it as you like, or leave it blank for non-interactive install
```

**Reply by SchoolGuy** [reply]
My bad. I meant `sample.ks`.

**Reply by 1392273211** [reply]
is %post
$SNIPPETS('post_anamon')
%end this content? I have also added this, but no logs are generated. Under /var/log/cobbler/anamon/, can the community improve it? Install cobbler on the client to check its installation progress.

**Reply by SchoolGuy** [reply]
Do you have the following setting enabled? - https://cobbler.readthedocs.io/en/latest/cobbler-conf/settings-yaml.html#anamon-enabled
